docs/notebooks/perturbed_optimizers/perturbed_optimizers.md

```diff
@@ -0,0 +1,380 @@
+---
+jupytext:
+  formats: ipynb,md:myst
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.14.4
+kernelspec:
+  display_name: base
+  language: python
+  name: python3
+---
+
+Copyright 2022 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at
+
+ https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+
++++
+
+# Perturbed optimizers
+
++++
+
+[![Open in Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/google/jaxopt/blob/main/docs/notebooks/perturbed_optimizers/perturbed_optimizers.ipynb)
+
++++
+
+We propose in this work a universal method to transform any optimizer $y^*$ in a differentiable approximation $y_\varepsilon^*$, using pertutbations following the method of [Berthet et al. (2020)](https://arxiv.org/abs/2002.08676). We provide a JAX implementation, illustrated here on some examples.
+
+Concretely, for an optimizer function $y^*$ defined by
+
+$$y^*(\theta) = \mathop{\mathrm{arg\,max}}_{y\in \mathcal{C}} \langle y, \theta \rangle$$
+
+We consider, for a random $Z$ drawn from a distribution with continuous positive distribution $\mu$
+
+$$y_\varepsilon^*(\theta) = \mathbf{E}[\mathop{\mathrm{arg\,max}}_{y\in \mathcal{C}} \langle y, \theta + \varepsilon Z \rangle]$$
+
+```{code-cell} ipython3
+:id: 9WIWwRdSU51j
+
+%%capture
+%pip install jaxopt
+```
+
+```{code-cell} ipython3
+:id: S6tLyyy9VCEw
+
+import jax
+import jax.numpy as jnp
+import jaxopt
+import time
+
+from jaxopt import perturbations
+```
+
++++ {"id": "EmYn_jNUFfw2"}
+
+# Argmax one-hot
+
++++
+
+We start from an original optimizer, an `argmax_one_hot` function, corresponding to $y^*$ for $\mathcal{C}$ being the unit simplex. We run it on an example input `values`
```

Comment:
What does "original" mean in this context? It is not clear to me

Reply:
I think it would also be good to explain with words what the `argmax_one_hot` does. Something along the lines of:

"We consider the `argmax_one_hot` function, which transforms a real-valued vector into a binary vector with one in the coefficient with largest magnitude and zero elsewhere"

Reply:
done